include/hipSYCL/sycl/libkernel/detail/local_memory_allocator.hpp

```diff
@@ -157,7 +158,10 @@ class host_local_memory
   // for more local memory we go to the heap.
   static constexpr size_t _max_static_local_mem_size = 1024*32;
   inline static char* _local_mem;
-  inline static std::array<char, _max_static_local_mem_size> _static_local_mem;
+  
+  alignas(alignof(std::max_align_t)) inline static char _static_local_mem
```

Comment:
Any specific reason to not use `std::aligned_storage_t<_max_static_local_mem_size, alignof(std::max_align_t)> _static_local_mem{};`?

Reply:
I think `aligned_storage_t` does not define which underlying data type it uses apart from using a POD type. Using a char array directly here probably avoids some reinterpret pointer casts later on.